src/webapi/events/drag.rs

```diff
@@ -0,0 +1,723 @@
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "DragEvent")]
+#[reference(subclass_of(Event, UiEvent, MouseEvent, DragRelatedEvent))]
+pub struct DragDropEvent(Reference);
+
+impl IEvent for DragDropEvent {}
+
+impl IUiEvent for DragDropEvent {}
+
+impl IMouseEvent for DragDropEvent {}
+
+impl IDragEvent for DragDropEvent {}
+
+impl ConcreteEvent for DragDropEvent {
+    const EVENT_TYPE: &'static str = "drop";
+}
+
+/// The dragend event is fired when a drag operation is being ended (by releasing a mouse button or
+/// hitting the escape key).
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/Events/dragend)
+// https://www.w3.org/TR/html51/editing.html#eventdef-global-dragend
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "DragEvent")]
+#[reference(subclass_of(Event, UiEvent, MouseEvent, DragRelatedEvent))]
+pub struct DragEndEvent(Reference);
+
+impl IEvent for DragEndEvent {}
+
+impl IUiEvent for DragEndEvent {}
+
+impl IMouseEvent for DragEndEvent {}
+
+impl IDragEvent for DragEndEvent {}
+
+impl ConcreteEvent for DragEndEvent {
+    const EVENT_TYPE: &'static str = "dragend";
+}
+
+/// The DataTransfer object is used to hold the data that is being dragged during a drag and drop
+/// operation.
+///
+/// It may hold one or more data items, each of one or more data types.
+/// For more information about drag and drop, see HTML Drag and Drop API.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer)
+// https://www.w3.org/TR/html51/editing.html#datatransfer-datatransfer
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "DataTransfer")]
+pub struct DataTransfer( Reference );
+impl DataTransfer {
+    /// Gets the type of drag-and-drop operation currently selected type.
+    /// The value must be none, copy, link or move.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/dropEffect)
+    // https://www.w3.org/TR/html51/editing.html#ref-for-dom-datatransfer-dropeffect-2
+    pub fn drop_effect( &self ) -> DropEffect {
+        let v: String = js!(
+            return @{self.as_ref()}.dropEffect;
+        ).try_into().unwrap();
+        match v.as_ref() {
+            "copy" => DropEffect::Copy,
+            "move" => DropEffect::Move,
+            "link" => DropEffect::Link,
+            "none" => DropEffect::None,
+            other => panic!("Expected valid dropEffect value, got {:?}", other),
+        }
+    }
+
+    /// Sets the type of drag-and-drop operation currently selected.
+    /// The value must be none, copy, link or move.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Element/dropEffect)
+    // https://www.w3.org/TR/html51/editing.html#ref-for-dom-datatransfer-dropeffect-2
+    pub fn set_drop_effect( &self, value: DropEffect ) {
```

Comment:
If we're going to add the other variant (which will contain a `String` internally) then this should be taken as `&DropEffect`. (Since the type won't be `Copy` anymore.)

Reply:
See my comment in the other thread: after looking into it, the `Other` variant was only needed for `DataTransferItemKind` so I don't think this comment applies anymore